tests/tests_relp_server.yml

```diff
@@ -12,14 +12,61 @@
 - name: Test the server configuration containing tls tcp, plain tcp and udp connection
   hosts: all
   vars:
-    __test_ca_cert_name: ca.pem
-    __test_cert_name: test-cert.pem
+    __test_ca_key_name: test-ca-key.pem
+    __test_ca_csr: test-ca.csr
+    __test_ca_cert_name: test-ca.crt
     __test_key_name: test-key.pem
-    __test_ca_cert: "{{ __test_ca_cert_name }}"
-    __test_cert: "{{ __test_cert_name }}"
-    __test_key: "{{ __test_key_name }}"
+    __test_csr_name: test-cert.csr
+    __test_cert_name: test-cert.pem
+    __test_ca_key: "/etc/pki/tls/private/{{ __test_ca_key_name }}"
+    __test_ca_cert_csr: "/etc/rsyslog.d/{{ __test_ca_csr }}"
+    __test_ca_cert: "/etc/pki/tls/certs/{{ __test_ca_cert_name }}"
+    __test_key: "/etc/pki/tls/private/{{ __test_key_name }}"
+    __test_cert_csr: "/etc/rsyslog.d/{{ __test_csr_name }}"
+    __test_cert: "/etc/pki/tls/certs/{{ __test_cert_name }}"
 
   tasks:
+    - name: Generate a CA key
+      openssl_privatekey:
+        path: "{{ __test_ca_key }}"
+        type: RSA
+
+    - name: Generate a CA Certificate Signing Request
+      openssl_csr:
+        path: "{{ __test_ca_cert_csr }}"
+        privatekey_path: "{{ __test_ca_key }}"
+        common_name: test-ca.system_roles.fedora.org
+
+    - name: Generate a self signed CA cert using the CA key
+      openssl_certificate:
+        path: "{{ __test_ca_cert }}"
+        privatekey_path: "{{ __test_ca_key }}"
+        csr_path: "{{ __test_ca_cert_csr }}"
+        provider: selfsigned
+
+    - name: Generate a server/client key
+      openssl_privatekey:
+        path: "{{ __test_key }}"
+        type: RSA
+
+    - name: Generate a server Certificate Signing Request
+      openssl_csr:
+        path: "{{ __test_cert_csr }}"
+        privatekey_path: "{{ __test_key }}"
+        country_name: US
+        organization_name: Fedora
+        email_address: logging@fedora.org
+        common_name: logging.fedora.org
+
+    - name: Generate a server/client cert using the key, and signed by the self signed CA
+      openssl_certificate:
+        path: "{{ __test_cert }}"
+        privatekey_path: "{{ __test_key }}"
+        csr_path: "{{ __test_cert_csr }}"
+        ownca_path: "{{ __test_ca_cert }}"
+        ownca_privatekey_path: "{{ __test_ca_key }}"
+        provider: ownca
```

Comment:
Nice.
This should be refactored into something like `tests/tasks/create_tests_certs.yml` for reuse.